src/main/java/hudson/plugins/powershell/PowerShell.java

```diff
@@ -26,7 +35,11 @@ protected String getFileExtension() {
 
     public String[] buildCommandLine(FilePath script) {
         if (isRunningOnWindows(script)) {
-            return new String[] { "powershell.exe", "-NonInteractive", "-ExecutionPolicy", "Bypass", "& \'" + script.getRemote() + "\'"};
+            if (useProfile){
```

Comment:
I was just thinking about adding this very feature.  Great to see you have already submitted a PR for it.  Any chance you could apply the same logic to the non-Windows case?  The same principle applies there.  If you rebased you'd pick up my fix to line 46 `powershell` -> `pwsh`.  Thanks!